tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -30,48 +31,47 @@ class ReinforceAgent:
     # pylint: disable=too-many-instance-attributes
     def __init__(
         self,
-        id,
-        obs_dim,
-        act_dim,
-        epsilon=0.01,
-        gamma=0.95,
+        id="*",
+        obs_dim=4,
+        act_dim=2,
+        gamma=0.99,
         lr=3e-4,
         max_replay_buffer_size=50000,
-        seed=42,
-        epsilon_schedule=None,
+        seed=21,
         lr_schedule=None,
+        model_params=None
     ):
-        self.id = id
+
         self._params = {}
+        self._params["id"] = id
         self._params["obs_dim"] = obs_dim
         self._params["act_dim"] = act_dim
         self._params["seed"] = seed
         self._params["max_replay_buffer_size"] = max_replay_buffer_size
         self._params["gamma"] = gamma
+        self._params["lr"] = lr
+        self._params["_lr_schedule"] = lr_schedule
+        self._params["model_params"] = model_params
+
+        self.init_agent()
 
-        self._epsilon_schedule = epsilon_schedule
-        if epsilon_schedule is None:
-            self._epsilon_schedule = ConstantSchedule(epsilon)
+    def init_agent(self):
 
-        self._lr_schedule = lr_schedule
-        if lr_schedule is None:
-            self._lr_schedule = ConstantSchedule(lr)
+        if self._params["_lr_schedule"] is None:
+            self._params["_lr_schedule"] = ConstantSchedule(self._params["lr"])
 
         self._model = PolicyNetwork(self._params["obs_dim"], self._params["act_dim"])
-        self._optimizer = tf.keras.optimizers.Adam(learning_rate=self._lr_schedule.get_value())
+        self._optimizer = tf.keras.optimizers.Adam(learning_rate=self._params["_lr_schedule"].get_value())
 
-        self._replay_buffer = CircularReplayBuffer(
-            seed=self._params["seed"], size=self._params["max_replay_buffer_size"]
-        )
+        if self._params["model_params"] is not None:
```

Comment:
This syntax is usually preferred (and will catch weird settings of the value to None which shouldn't happen.

```suggestion
        if "model_params" in self._params
```